apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -35,7 +36,33 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         public string DataSource
         {
             get => GetValueOrDefault("Data Source");
-            private set => this["Data Source"] = value;
+            private set => this["Data Source"] = ValidatedDataSource(value);
+        }
+
+        private string ValidatedDataSource(string dataSource)
```

Comment:
dont think case sensitivity is that important.
hmm good point re:using templates -- except for the case where you do CREATE DATABASE and do not have a database.
i was aiming to allow you to only specify the first two...  Let me see what I can do here.